ProjFS.Mac/PrjFSLib/prjfs-log/prjfs-log.cpp

```diff
@@ -39,15 +41,23 @@ int main(int argc, const char * argv[])
             {
                 break;
             }
+            
             int messageSize = entry->size;
             if (messageSize >= sizeof(KextLog_MessageHeader) + 2)
             {
                 struct KextLog_MessageHeader message = {};
                 memcpy(&message, entry->data, sizeof(KextLog_MessageHeader));
                 const char* messageType = KextLogLevelAsString(message.level);
                 int logStringLength = messageSize - sizeof(KextLog_MessageHeader) - 1;
-                printf("%s: %.*s\n", messageType, logStringLength, entry->data + sizeof(KextLog_MessageHeader));
+
+                struct timeval tp;
+                gettimeofday(&tp, NULL);
+                long int timestamp = tp.tv_sec * 1000 + tp.tv_usec / 1000;
+                
+                printf("(%d: %ld) %s: %.*s\n", lineCount, timestamp, messageType, logStringLength, entry->data + sizeof(KextLog_MessageHeader));
```

Comment:
@sanoursa, what was your reason for adding line number to the output?

Reply:
It just made it easier for me to compare different test runs and compare logs. I added the timestamp later, you could definitely argue that we don't need both. What are your thoughts?

Reply:
I had been thinking we wouldn't need line numbers (text editors can provide that) but I can seen the benefit now to keeping the line numbers because we expect the log message to contain new lines.

I think we can keep the line numbers in

Reply:
Yeah, I can see the value in keeping the line numbers. I vote to keep them in too 👍

Reply:
Now that you mention it, I think that (multiline traces) is why I added it :-). Hard to remember to the way back from before vacation.

Reply:
For what it's worth, there is also a timestamp field in the message itself. This is in units of Mach Absolute Time, which is a monotonically increasing clock as opposed to one linked to system time. This time stamp of course also reflects the time when the kernel generated the log message, not when it arrived in the logging daemon. Given we're just outputting seconds here, using the Mach timestamp may be closer to what we want, although I'm of course not exactly sure what it is you do want. One major difference would be that it's not in seconds since midnight, but (nano-)seconds since some arbitrary time, such as the time when the logging daemon started.

Reply:
Ah it probably makes sense to use that then. I just wanted to be able to see relative timings of events. I'll clean that up before merging.